value.go

```diff
@@ -0,0 +1,20 @@
+package gostub
+
+// TestingT is a subset of the testing.TB interface used by gostub.
+type TestingT interface {
+	Cleanup(func())
+}
+
+// Value replaces the value at varPtr with stubVal.
+// The original value is reset at the end of the test via t.Cleanup
+// or can be reset using the returned function.
+func Value[T any](t TestingT, varPtr *T, stubVal T) (reset func()) {
```

Comment:
Should there only be a single function that always takes `TestingT`, or should there be 2 versions -- `Value` that doesn't take `TestingT` and doesn't auto-reset, and `ValueT` which takes `TestingT` and auto-resets

Reply:
Going with `Value` for now, since auto-reset is a better default for callers, so it should have less noise.